akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala

```diff
@@ -524,6 +524,13 @@ private[akka] class RemoteActorRefProvider(
    * public `resolveActorRef(path: String)`.
    */
   private[akka] def internalResolveActorRef(path: String): ActorRef = path match {
+
+    case p if IgnoreActorRef.isIgnoreRefPath(p) =>
+      // we need a local address to avoid sending messages that will be ignored over the wire
+      // basically, we pretend that this Ref belongs to this system
+      val localAddressToUse = transport.localAddressForRemote(systemGuardian.path.address)
+      new IgnoreActorRef(localAddressToUse.hostPort)
```

Comment:
This is the trick that guarantees that a IgnoreActorRef that is deserialized on another node, won't send unnecessary messages over the wire. We pretend it's a local Actor.